web/react/components/popover_list_members.jsx

```diff
@@ -27,12 +43,62 @@ export default class PopoverListMembers extends React.Component {
             }
         };
     }
+
+    handleShowDirectChannel(teammate, e) {
+        e.preventDefault();
+
+        const channelName = Utils.getDirectChannelName(UserStore.getCurrentId(), teammate.id);
+        let channel = ChannelStore.getByName(channelName);
+
+        const preference = PreferenceStore.setPreference(Constants.Preferences.CATEGORY_DIRECT_CHANNEL_SHOW, teammate.id, 'true');
+        AsyncClient.savePreferences([preference]);
+
+        if (channel) {
+            Utils.switchChannel(channel);
+            this.closePopover();
+        } else {
+            channel = {
+                name: channelName,
+                last_post_at: 0,
+                total_msg_count: 0,
+                type: 'D',
+                display_name: teammate.username,
+                teammate_id: teammate.id,
+                status: UserStore.getStatus(teammate.id)
+            };
+
+            Client.createDirectChannel(
+                channel,
+                teammate.id,
+                (data) => {
+                    AsyncClient.getChannel(data.id);
+                    Utils.switchChannel(data);
+
+                    this.closePopover();
+                },
+                () => {
+                    window.location.href = TeamStore.getCurrentTeamUrl() + '/channels/' + channelName;
+                    this.closePopover();
+                }
+            );
+        }
+    }
+
+    closePopover() {
+        var overlay = this.refs.overlay;
```

Comment:
Instead of refs and calling methods on children, passing props to child components should be used when possible. Here we can use the `show` property of OverlayTrigger and the `onHide` callback to keep the show state up to date. Details can be found here: https://react-bootstrap.github.io/components.html#overlays